### PR TITLE
Implemented DiningAPI

### PR DESCRIPTION
**Pull request by Rahi374**
I put most of the information in the readme (I hope you don't mind that I added DiningAPI examples and TODOs).

We now have:
* `get_dining_locations` - returns a dict of dicts all the dining locations, each dict with the name, open/closed status, and the open times
* `get_dining_locations_by_status(status=None)` - same as `get_dining_locations()`, but filter by open/closed status (`status="open"` or `status="closed"`). In fact, `get_dining_locations()` just calls this with `status=None`
* `get_dining_location_by_name(location)` - returns a single dict of the requested dining location with its name, open/closed status, and the open times
* `@staticmethod _encode_dining_location(string)` - inputs the full name of the dining location and turns it into a key usable with the dicts
* `@staticmethod _decode_dining_location(string)` - inputs the key usable with the dicts and turns it back into the full name of the dining location

Still need to implement `get_dining_location_menu(location)`, but the above implemented methods cover the originally requested functionalities for the DiningAPI.

### CODE REVIEW

File: pittAPI.py [hunk 1]
```diff
@@ -428,3 +429,123 @@ def get_person(self, query, maxPeople=10):
...
+        
+        #sample_url = "https://m.pitt.edu/dining/index.json?_region=kgoui_Rcontent_I1_Ritems&_object_include_html=1&_object_js_config=1&_kgoui_page_state=eb95bc72eca310cbbe76a39964fc7143&_region_index_offset=15&feed=dining_locations&start=15"
+        # the _region_index_offset is optional
+        # -- seems like it's only for the id of the li for the html
+        dining_locations = {}
+        
+        end_loop = False
+        load_more = False
+        counter = 0
+        while( not end_loop ):
```
Comment (RitwikGupta) [inline pittAPI.py]: Parentheses are not needed

File: pittAPI.py [hunk 2]
```diff
@@ -428,3 +429,123 @@ def get_person(self, query, maxPeople=10):
...
+            load_more = False
+            url = "https://m.pitt.edu/dining/index.json?_region=kgoui_Rcontent_I1_Ritems&_object_include_html=1&_object_js_config=1&_kgoui_page_state=eb95bc72eca310cbbe76a39964fc7143&feed=dining_locations&start=" + str(counter)
+            data = json.load(urlopen(url))
+            soup = BeautifulSoup(data['response']['html'], 'html.parser')
+            res = soup.find_all("div", class_="kgoui_list_item_textblock")
+
+            for i in res:
+                if( i.find("span").getText() != "Load more..." ):
```
Comment (RitwikGupta) [inline pittAPI.py]: Stylistically, get rid of these parentheses across all statements. You can use parens for complicated inner logic wraps but there isn't one required like `if ()`

File: pittAPI.py [hunk 3]
```diff
@@ -428,3 +429,123 @@ def get_person(self, query, maxPeople=10):
...
+                            dining_locations[self._encode_dining_location(i.find("span").getText())] = {
+                                "name"   : i.find("span").getText(),
+                                "hours"  : i.find("div").getText().replace(u'\u2013', '-').replace('\n', '').replace('Next: ', ''),
+                                "status" : "closed" if "Next:" in i.find("div").getText() else "open"
+                            }
+                            end_loop = True
+                    elif(status != "open"):
+                        #dining_locations.append([i.find("span").getText(), ""])
+                        dining_locations[self._encode_dining_location(i.find("span").getText())] = {
+                            "name"   : i.find("span").getText(),
```
Comment (RitwikGupta) [inline pittAPI.py]: There should be no spaces between a key and it's corresponding `:` according to pep8. Get rid of all key-colon spaces across the code.

Comment (Rahi374) [inline pittAPI.py]: I wanted to line up the dict values but okay I'll fix it.

**Comment by Rahi374** [comment]
I'll get rid of the unit tests then

**Comment by RitwikGupta** [comment]
I think you should keep them, just use `unittest` to implement them. We **really** need tests so this is super helpful in any way, shape, or form 😄

File: pittAPI.py [hunk 4]
```diff
@@ -472,24 +472,24 @@ def get_dining_locations_by_status(self, status=None):
                             #    i.find("div").getText().replace(u'\u2013', '-').replace('\n', '')
                             #])
                             dining_locations[self._encode_dining_location(i.find("span").getText())] = {
-                                "name"   : i.find("span").getText(),
-                                "hours"  : i.find("div").getText().replace(u'\u2013', '-').replace('\n', '').replace('Next: ', ''),
-                                "status" : "closed" if "Next:" in i.find("div").getText() else "open"
+                                "name":i.find("span").getText(),
```
Comment (RitwikGupta) [inline pittAPI.py]: Last minor nitpick then this is good to merge, but there should be a space after the colon between it and the value!

Comment (Rahi374) [inline pittAPI.py]: It's funny because that's actually what I wanted to do :smile:

File: pittAPI.py [hunk 5]
```diff
@@ -428,3 +429,123 @@ def get_person(self, query, maxPeople=10):
...
+        # -- seems like it's only for the id of the li for the html
+        dining_locations = {}
+        
+        end_loop = False
+        load_more = False
+        counter = 0
+        while not end_loop :
+            load_more = False
+            url = "https://m.pitt.edu/dining/index.json?_region=kgoui_Rcontent_I1_Ritems&_object_include_html=1&_object_js_config=1&_kgoui_page_state=eb95bc72eca310cbbe76a39964fc7143&feed=dining_locations&start=" + str(counter)
+            data = json.load(urlopen(url))
```
Comment (RitwikGupta) [inline pittAPI.py]: `NameError: global name 'urlopen' is not defined`
We're using `requests` now, not urllib

File: pittAPI.py [hunk 6]
```diff
@@ -495,7 +495,10 @@ def get_dining_locations_by_status(self, status=None):
         return dining_locations
 
     def get_dining_location_by_name(self, location):
-        return self.get_dining_locations()[self._encode_dining_location(location)]
+        try:
+            return self.get_dining_locations()[self._encode_dining_location(location)]
+        except:
+            return {}
```
Comment (RitwikGupta) [inline pittAPI.py]: I would throw an error. Silent failures aren't fun and Exceptions exist to inform the programmer they messed up somewhere. Sorry for the long review! This is some really high quality work, there's just a lot of it.

Comment (Rahi374) [inline pittAPI.py]: Should I throw a custom exception? -- oh wait we have the 'InvalidParameterException'

Comment (RitwikGupta) [inline pittAPI.py]: Yes, InvalidParameterException!

**Comment by RitwikGupta** [comment]
Closed #18!

**Pull request merged**